testah-junit/src/main/java/org/testah/framework/testPlan/TestSystem.java

```diff
@@ -0,0 +1,608 @@
+package org.testah.framework.testPlan;
+
+import org.junit.Assume;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.jupiter.api.Disabled;
+import org.junit.runner.Description;
+import org.slf4j.MDC;
+import org.testah.TS;
+import org.testah.client.dto.StepActionDto;
+import org.testah.client.dto.TestCaseDto;
+import org.testah.client.dto.TestPlanDto;
+import org.testah.client.dto.TestStepDto;
+import org.testah.framework.annotations.KnownProblem;
+import org.testah.framework.annotations.TestCase;
+import org.testah.framework.annotations.TestPlan;
+import org.testah.framework.cli.Cli;
+import org.testah.framework.cli.TestFilter;
+import org.testah.framework.dto.TestCaseAnnotationDto;
+import org.testah.framework.dto.TestDtoHelper;
+import org.testah.framework.dto.TestPlanAnnotationDto;
+import org.testah.runner.TestahJUnitRunner;
+import org.testah.runner.testPlan.TestPlanActor;
+
+import java.lang.reflect.Method;
+import java.util.HashMap;
+
+public class TestSystem {
+
+    /**
+     * The test plan.
+     */
+    protected ThreadLocal<TestPlanDto> testPlan = new ThreadLocal<>();
+
+    /**
+     * The test case.
+     */
+    protected ThreadLocal<TestCaseDto> testCase = new ThreadLocal<>();
+
+    /**
+     * The test step.
+     */
+    protected ThreadLocal<TestStepDto> testStep = new ThreadLocal<>();
+
+    /**
+     * The test plan start.
+     */
+    protected ThreadLocal<Boolean> testPlanStart = new ThreadLocal<>();
+
+    /**
+     * The test filter.
+     */
+    public TestFilter testFilter = null;
+
+    /**
+     * The ignored tests.
+     */
+    protected ThreadLocal<HashMap<String, String>> ignoredTests = null;
+
+    /**
+     * The description.
+     */
+    private Description description;
+
+    /**
+     * Starting of a testCase.
+     *
+     * @param description testCase description sent by junit.
+     */
+    public void starting(final Description description) {
+        starting(description,
+            TestPlanAnnotationDto.create(description.getTestClass().getAnnotation(TestPlan.class)),
+            TestCaseAnnotationDto.create(description.getAnnotation(TestCase.class)));
+    }
+
+    /**
+     * Starting of a testCase.
+     *
+     * @param desc testCase description sent by junit.
+     * @param testPlan testPlan annotation info.
+     * @param testCase testCase annotation info.
+     */
+    public void starting(final Description desc, TestPlanAnnotationDto testPlan, TestCaseAnnotationDto testCase) {
+        setDescription(desc);
+        if (!didTestPlanStart()) {
+
+            setUpThreadLocals();
+
+            TS.log().info("TESTPLAN started:" + desc.getTestClass().getName() + " - thread[" + Thread.currentThread().getId() + "]");
+            if (null == testPlan) {
```

Comment:
Test plan is optional? All you need is a test case?

Reply:
It was for junit 4, as the heavy lifting was in the extended class, but for junit 5 it really is required.  In order to not introduce a breaking change I will make the starting aware if it is junit 4 or 5, if 5 then this will throw a runtime exception.